scripts/keys.coffee

```diff
@@ -10,95 +10,247 @@
+				users = robot.brain.userForName othername
+				if users is null
+					key_holder = null
 				else
-					msg.send "#{othername}? Never heard of 'em"
-
-	robot.respond /(i|I) (have given|gave|had given) (the key|key|keys|a key|the keys) to (.+)/i , (msg)->
-		othername = msg.match[4]
+					key_holder = users.name
+				if key_holder is null
+					if othername is 'you'
+						msg.send "How am I supposed to take those #{keyname} keys? #{name} is a liar!"
+					else if othername is robot.name
+						msg.send "That's utter lies! How can you blame a bot to have the keys?"
+					else
+						msg.send "I don't know anyone by the name #{othername}"
+				else
+					if keyname is 'red'
+						robot.brain.set("keyRed",key_holder)
+					else if keyname is 'blue'
+						robot.brain.set("keyBlue",key_holder)
+					else if keyname is 'yellow'
+						robot.brain.set("keyYellow",key_holder) 
+					else if keyname is 'green'
+						robot.brain.set("keyGreen",key_holder) 
+					else if keyname is 'orange'
+						robot.brain.set("keyOrange",key_holder) 
+					msg.send "Okay, so now the #{keyname} keys are with #{othername}"	
+    #4this section belongs to "i gave the keys to $name"
+	robot.respond /(i|I) (have given the|had given the|have given|gave the|had given|gave) (.+) (key|keys) to (.+)/i , (msg)->
+		othername = msg.match[5]
+		keyname = msg.match[3]
 		name = msg.message.user.name
-		k = key()
-		if othername is 'you'
-			msg.send "That's utter lies! How can you blame a bot to have the keys?"
-		else if othername is robot.name
-			msg.send "That's utter lies! How can you blame a bot to have the keys?"
-		else
+		unless keyname not in ["red", "blue", "yellow", "green", "orange"]
 			users = robot.brain.userForName othername
 			if users is null
-				msg.send "I don't know anyone by the name #{othername}"
+				key_holder = null
 			else
-				k = users.name
-				msg.send "Okay, so now the keys are with #{users.name}"		
-
-		robot.brain.set("key",k)		
-				
-	robot.respond /(who|who all) (has|have) (the key|key|keys|a key)/i , (msg)->
-		k = key()
-		msgText = k
+				key_holder = users.name
+			if key_holder is null
+				if othername is 'you'
+					msg.send "That's utter lies! How can you blame a bot to have the keys? #{name} is a liar!"
+				else if othername is robot.name
+					msg.send "That's utter lies! How can you blame a bot to have the keys?"
+				else
+					msg.send "I don't know anyone by the name #{othername}"
+			else
+				if keyname is 'red'
+					robot.brain.set("keyRed",key_holder)	
+				else if keyname is 'blue'
+					robot.brain.set("keyBlue",key_holder)
+				else if keyname is 'yellow'
+					robot.brain.set("keyYellow",key_holder)
+				else if keyname is 'green'
+					robot.brain.set("keyGreen",key_holder)
+				else if keyname is 'orange'
+					robot.brain.set("keyOrange",key_holder)
+				msg.send "Okay, so now the #{keyname} keys are with #{users.name}"
+	
+	#5this section is to print the details about the key holder.		
```

Comment:
What will happen if I enter "Who has keys" ?

Reply:
code From line no. 225 executes

Reply:
Type a sample output

Reply:
oh ok.There is an edit needed in line 242,245.
after edit->
bro who has the keys
[name] has red keys
..
..
..
***********************
bro who has keys
Ah, nobody informed.....